library/dnet.c

```diff
@@ -499,9 +501,26 @@ static int dnet_iterator_server_send_complete(struct dnet_addr *addr, struct dne
 			if (!re->status)
 				re->status = err;
 
-			dnet_log(st->n, DNET_LOG_NOTICE, "%s: %s: sending response to client: %s, "
+			if (!err && !re->status) {
+				long usec;
+				struct timeval tv;
+
+				gettimeofday(&tv, NULL);
+
+				usec = (tv.tv_sec - wp->start.tv_sec) * 1000000 + (tv.tv_usec - wp->start.tv_usec);
+
+				/*
+				 * Maximum number of bytes written into the wire and not yet acknowledged.
+				 * It is equal to 'current speed' times 5 seconds, since 5 seconds is default wait timeout
+				 * for write command.
+				 */
+				new_pending = 5 * re->size * 1000000 / usec;
```

Comment:
5 seconds is our default timeout, I used this number for upper boundary, this is a heuristic number, which allows iterator to actually copy data over 1gbit network between common disks without timeouts, in theory it might be higher, but there is a risk of overfilling the disk bandwidth.